src/library/modules/OpenDBSubmission.js

```diff
@@ -0,0 +1,218 @@
+		state: null,
+		createShipData: null,
+		dropShipData: null,
+
+		// api handler
+		handlers: {},
+		// <map id> => <event rank>
+		mapInfo: {},
+
+		init: function (){
+			this.handlers = {
+				'api_get_member/mapinfo': this.processMapInfo,
+				'api_req_kousyou/createship': this.processCreateShip,
+				'api_get_member/kdock': this.processKDock,
+				'api_req_kousyou/createitem': this.processCreateItem,
+				'api_req_kousyou/remodel_slot': this.processRemodelItem,
+				'api_req_map/select_eventmap_rank': this.processSelectEventMapRank,
+				// start or next
+				'api_req_map/start': this.processStartNext,
+				'api_req_map/next': this.processStartNext,
+
+				// detect ship id
+				'api_req_sortie/battleresult': this.processBattleResult,
+				'api_req_combined_battle/battleresult': this.processBattleResult
+			};
+		},
+		processSelectEventMapRank: function (requestObj) {
+			var params = requestObj.params;
+			// I know they are strings, just being 100% sure.
+			var mapId = String(params.api_maparea_id) + String(params.api_map_no);
+			var rank = parseInt(params.api_rank, 10);
+			this.mapInfo[mapId] = rank;
+		},
+		processCreateShip: function ( requestObj ) {
+			this.cleanup();
+			var params = requestObj.params;
+			var createShipData = {
+				apiver: 2,
+				flagship: PlayerManager.fleets[0].ship(0).masterId,
+				fuel: parseInt(params.api_item1),
+				ammo: parseInt(params.api_item2),
+				steel: parseInt(params.api_item3),
+				bauxite: parseInt(params.api_item4),
+				material: parseInt(params.api_item5),
+				kdockId: parseInt(params.api_kdock_id) - 1
+			};
+			this.createShipData = createShipData;
+			this.state = "create_ship";
+		},
+		processKDock: function( requestObj ) {
+			if (this.state !== 'create_ship') {
+				this.cleanup();
+				return;
+			}
+			var createShipData = this.createShipData;
+			createShipData.result =
+				requestObj.response.api_data[createShipData.kdockId].api_created_ship_id;
+			delete createShipData.kdockId;
+
+			// console.log( "[createship] prepared: " + JSON.stringify( createShipData ) );
+			this.submitData("ship_dev.php", createShipData);
+			this.state = null;
+		},
+		processCreateItem: function( requestObj ) {
+			this.cleanup();
+			var params = requestObj.params;
+			var response = requestObj.response.api_data;
+			var createItemData = {
+				apiver: 2,
+				flagship: PlayerManager.fleets[0].ship(0).masterId,
+				fuel: params.api_item1,
+				ammo: params.api_item2,
+				steel: params.api_item3,
+				bauxite: params.api_item4,
+				result: (response.api_slot_item === undefined) ? 0 : response.api_slot_item.api_slotitem_id
+			};
+
+			// console.log( "[createitem] prepared: " + JSON.stringify( createItemData ));
+			this.submitData("equip_dev.php", createItemData);
+		},
+		processStartNext: function( requestObj ) {
+			this.cleanup();
+			if(KC3ShipManager.count() >= KC3ShipManager.max || KC3GearManager.count() >= KC3GearManager.max)
```

Comment:
yep... it's updated at another unmerged branch 😂 
https://github.com/KC3Kai/KC3Kai/blob/dev/meta-live-reload/src/library/managers/GearManager.js#L12